assets/js/controllers/settings/manageSecondPassword.controller.js

```diff
@@ -61,4 +69,44 @@ function ManageSecondPasswordCtrl ($rootScope, $scope, Wallet, $timeout, MyWalle
     $scope.status.waiting = true;
     Wallet.setSecondPassword($scope.fields.password, success);
   };
+
+  $scope.recoveryModal = () => {
+    const openModal = () => $uibModal.open({
+      templateUrl: 'partials/recovery-before-second-password.jade',
+      controller: 'RecoveryBeforeSecondPasswordCtrl',
+      windowClass: 'bc-modal'
+    });
+
+    if (!Wallet.status.didConfirmRecoveryPhrase && !Wallet.settings.secondPassword) {
+      openModal();
+    }
+  };
+
+  $scope.showButton = () => {
+    const confirmed = $scope.walletStatus.didConfirmRecoveryPhrase;
+    const dismissed = $scope.walletStatus.dismissedRecoveryPrompt;
+    const secondPW = Wallet.settings.secondPassword;
+
+    if (confirmed) {
+      return true;
+    } else if (dismissed && secondPW) {
+      return true;
+    }
+  };
+
+  $scope.hideButton = () => {
+    const confirmed = $scope.walletStatus.didConfirmRecoveryPhrase;
+    const dismissed = $scope.walletStatus.dismissedRecoveryPrompt;
+    const secondPW = Wallet.settings.secondPassword;
+
+    if (confirmed && !secondPW) {
+      return false;
+    } else if (secondPW) {
+      return true;
+    } else if (!dismissed || !confirmed) {
+      return true;
+    } else if (dismissed && secondPW) {
+      return true;
+    }
+  };
```

Comment:
I mentioned this in a previous comment, but I think this show / hide logic could be simplified a lot by having a single button, and really considering what it's purpose is.